LokiLabs/microsoft-graph-devx-api
# AB#38606 graph requests

## Overview

* Be able to run queries as an application and use resource-specific consent
* This is part of the proxy work flow
* Built off of #11 because this requires the token

### Demo


https://user-images.githubusercontent.com/47228306/126374095-51bef39d-5eb6-425b-9ca0-fa4cf140e079.mp4


## Testing Instructions
* Pull this branch
* Fill in the TenantId, ClientId, ClientSecret and AzureConnectionString inside `appsettings.json` (Note that you need to have the necessary rsc permissions in the app to run the queries)
* Open in Visual Studio
* Run in IIS Express
* The endpoint is `https://localhost:44399/graphproxy/{tenantId}/{graph api call suffix}`
* What you add to your api call is what is passed to `https://graph.microsoft.com/`. Eg. GET `https://localhost:44399/graphproxy/{tenantId}/v1.0/` will be calling GET `https://graph.microsoft.com/v1.0/ .
